File: src/main/java/com/chtl/lexer/CHTLToken.java
package com.chtl.lexer;

public class CHTLToken {
    private String type;
    private String value;
    public CHTLToken(String type, String value) {
        this.type = type;
        this.value = value;
    }
    public String getType() { return type; }
    public String getValue() { return value; }
}

File: CHTLCompiler/src/lexer/CHTLToken.cpp
#include "lexer/CHTLToken.h"
#include <sstream>

namespace chtl {

std::string Token::toString() const {
    std::ostringstream oss;
    oss << "Token{type=" << static_cast<int>(type) 
        << ", value='" << value << "'"
        << ", line=" << line 
        << ", column=" << column 
        << ", position=" << position << "}";
    return oss.str();
}

std::string CHTLJSToken::toString() const {
    std::ostringstream oss;
    oss << "CHTLJSToken{type=" << static_cast<int>(type) 
        << ", value='" << value << "'"
        << ", line=" << line 
        << ", column=" << column 
        << ", position=" << position << "}";
    return oss.str();
}

} // namespace chtl